util/dockerfiles/docker-bake.hcl

```diff
@@ -35,59 +35,147 @@ variable "TAG" {
+    "clang-version-17",
+    "clang-version-18"
+  ]
 }
 
-group "clang-compilers" {
-  targets = ["clang-compilers-base-20-04", "clang-compilers-base-22-04", "clang-compilers-16"]
+target "clang-version-14" {
+  inherits = ["common"]
+  dockerfile = "Dockerfile"
+  args = {
+    version = "14"
+  }
+  context = "clang-compiler"
+  tags = ["${IMAGE_URI}/clang-version-14:${TAG}"]
 }
 
-group "gcc-compilers" {
-  targets = ["gcc-compilers-base-20-04", "gcc-compilers-base-22-04"]
+target "clang-version-15" {
+  inherits = ["common"]
+  dockerfile = "Dockerfile"
+  args = {
+    version = "15"
+  }
+  context = "clang-compiler"
+  tags = ["${IMAGE_URI}/clang-version-15:${TAG}"]
 }
 
-# Common attributes across all targets. Note: these can be overwritten.
-target "common" {
-  # Here we are enabling multi-platform builds. We are compiling to both ARM
-  # amd X86.
-  platforms = ["linux/amd64", "linux/arm64", "linux/riscv64"]
+target "clang-version-16" {
+  inherits = ["common"]
+  dockerfile = "Dockerfile"
+  args = {
+    version = "16"
+  }
+  context = "clang-compiler"
+  tags = ["${IMAGE_URI}/clang-version-16:${TAG}"]
 }
 
-target "gcn-gpu" {
+target "clang-version-17" {
   inherits = ["common"]
   dockerfile = "Dockerfile"
-  context = "gcn-gpu"
-  tags = ["${IMAGE_URI}/gcn-gpu:${TAG}"]
+  args = {
+    version = "17"
+  }
+  context = "clang-compiler"
+  tags = ["${IMAGE_URI}/clang-version-17:${TAG}"]
 }
 
-target "gpu-fs" {
+target "clang-version-18" {
   inherits = ["common"]
   dockerfile = "Dockerfile"
-  context = "gpu-fs"
-  tags = ["${IMAGE_URI}/gpu-fs:${TAG}"]
+  args = {
+    version = "18"
+  }
+  context = "clang-compiler"
+  tags = ["${IMAGE_URI}/clang-version-18:${TAG}"]
 }
 
-target "sst" {
+group "gcc-compilers" {
+  targets = [
+    "gcc-version-10",
+    "gcc-version-11",
+    "gcc-version-12",
+    "gcc-version-13"
+  ]
+}
+
+target "gcc-version-10" {
   inherits = ["common"]
   dockerfile = "Dockerfile"
-  context = "sst-11.1.0"
-  tags = ["${IMAGE_URI}/sst-env:${TAG}"]
+  args = {
+    version = "10"
+  }
+  context = "gcc-compiler"
+  tags = ["${IMAGE_URI}/gcc-version-10:${TAG}"]
 }
 
-target "systemc" {
+target "gcc-version-11" {
   inherits = ["common"]
   dockerfile = "Dockerfile"
-  context = "systemc-2.3.3"
-  tags = ["${IMAGE_URI}/systemc-env:${TAG}"]
+  args = {
+    version = "11"
+  }
+  context = "gcc-compiler"
+  tags = ["${IMAGE_URI}/gcc-version-11:${TAG}"]
+}
+
+target "gcc-version-12" {
+  inherits = ["common"]
+  dockerfile = "Dockerfile"
+  args = {
+    version = "12"
+  }
+  context = "gcc-compiler"
+  tags = ["${IMAGE_URI}/gcc-version-12:${TAG}"]
+}
+
+target "gcc-version-13" {
+  inherits = ["common"]
+  dockerfile = "Dockerfile"
+  args = {
+    version = "13"
+  }
+  context = "gcc-compiler"
+  tags = ["${IMAGE_URI}/gcc-version-13:${TAG}"]
+}
+
+group "ubuntu-releases" {
+  targets=[
+    "ubuntu-24-04_all-dependencies",
+    "ubuntu-22-04_all-dependencies",
+    "ubuntu-20-04_all-dependencies",
```

Comment:
Is this still supported now that we don't support gcc 8? Or, do we have something special in the dockerfile to upgrade GCC?

Reply:
Done.

Reply:
I suspect, but don't know for sure, that GPUSE will not work anymore if 20.04 is removed -- because the versions of gcc in newer versions don't work with ROCm 4.  But I guess we can run this test and see.